packages/client/src/types.ts

```diff
@@ -138,7 +138,24 @@ export interface SpendingLimitConfig {
 }
 
 export interface PrepareResponse {
-  transaction: string; // base64url encoded with Lighthouse instructions
+  transaction: string;
   simulationResult?: any;
   memoryConfigUsed?: SpendingLimitConfig;
 }
+
+export interface PrepareErrorResponse {
+  // RFC 7807 Problem Details format
+  type: string; // e.g., "spending-limit-exceeded", "transaction-blocked"
+  title: string;
+  detail: string;
+  requestId: string;
```

Comment:
We should move these to a separate type that is extended. A base class could then implement this as well.